lib/rules/hook-use-state.js

```diff
@@ -72,7 +72,7 @@ module.exports = {
       const expectedSetterVariableNames = upperCaseCandidatePrefix ? [
         `set${upperCaseCandidatePrefix.charAt(0).toUpperCase()}${upperCaseCandidatePrefix.slice(1)}${caseCandidateSuffix}`,
         `set${upperCaseCandidatePrefix.toUpperCase()}${caseCandidateSuffix}`,
-      ] : undefined;
```

Comment:
I usually try to avoid this empty-object fallback pattern as it allocates unnecessary empty objects, and because it aligns really naturally with the conditional-access-operator pattern in more-modern js versions.

Happy to follow with the suggestions, but just thought I'd share my thoughts, particularly because they've been shaped somewhat by this repository's tool-for-all-ages mentality.

Reply:
yeah, that's true, but without optional chaining, it's a lot grosser. i prefer having a constant type for things; it's simpler to reason about, and "object allocation" is a non-problem.